shared/teams/new-team/index.js

```diff
@@ -11,89 +11,152 @@ import {
+  pending: boolean,
+}
 
-const headerText = (baseTeam: string, errorText: string, isSubteam: boolean): string => {
-  if (errorText) {
-    return errorText
+class Contents extends React.Component<Props> {
+  _onSubmit = () => {
+    if (!validTeamnamePart(this.props.name)) {
+      this.props.onSetTeamCreationError(
+        'Teamnames must be between 2 and 16 characters long, can only contain letters and underscores, and cannot begin with an underscore.'
+      )
+      return
+    }
+    this.props.onSubmit(this._fullName())
   }
 
-  // TODO: Display an error and disable the submit button if name
-  // isn't a valid teamname.
-  return isSubteam
-    ? `You're creating a subteam of ${baseTeam}.`
-    : "For security reasons, team names are unique and can't be changed, so choose carefully."
-}
+  _errorText = () => {
+    if (this.props.errorText) {
+      return this.props.errorText
+    }
+    return null
+  }
 
-const Contents = ({
-  errorText,
-  baseTeam,
-  isSubteam,
-  joinSubteam,
-  name,
-  onJoinSubteamChange,
-  onNameChange,
-  onSubmit,
-  pending,
-}: Props) => (
-  <ScrollView>
-    <Box style={globalStyles.flexBoxColumn}>
-      <Box style={{...styleContainer, backgroundColor: errorText ? globalColors.red : globalColors.blue}}>
-        <Text
-          style={{margin: globalMargins.tiny, textAlign: 'center', width: '100%'}}
-          type="BodySemibold"
-          backgroundMode={errorText ? 'HighRisk' : 'Announcements'}
-        >
-          {headerText(baseTeam, errorText, isSubteam)}
-        </Text>
-      </Box>
+  _headerText = () => {
+    if (this.props.isSubteam) {
+      return `You are creating a subteam of ${this.props.baseTeam}`
+    }
+    return "For security reasons, team names are unique and can't be changed, so choose carefully."
+  }
 
-      <Box
-        style={{
-          ...globalStyles.flexBoxColumn,
-          ...stylePadding,
-          alignItems: 'center',
-          justifyContent: 'center',
-          backgroundColor: globalColors.white,
-        }}
-      >
-        <Box style={{...globalStyles.flexBoxRow, marginTop: globalMargins.medium}}>
-          <Input
-            autoFocus={true}
-            hintText="Name your team"
-            value={name}
-            onChangeText={onNameChange}
-            onEnterKeyDown={onSubmit}
-            disabled={pending}
-          />
-        </Box>
+  _fullName = () =>
+    this.props.isSubteam ? this.props.baseTeam.concat(`.${this.props.name}`) : this.props.name
 
-        {isSubteam && (
-          <Box style={{...globalStyles.flexBoxRow, marginTop: globalMargins.medium}}>
-            <Checkbox
-              checked={joinSubteam}
-              label="Join this subteam after creating it."
-              onCheck={onJoinSubteamChange}
-            />
+  render() {
+    const {isSubteam, joinSubteam, name, onJoinSubteamChange, onNameChange, pending} = this.props
+    const errorText = this._errorText()
+    return (
```

Comment:
Mostly the same, I just added the error text below the header text and the subteam name preview